mailet/api/src/main/java/org/apache/mailet/QueueSerializable.java

```diff
@@ -0,0 +1,48 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.mailet;
+
+import java.util.Optional;
+
+public interface QueueSerializable {
+    public class Serializable {
+        private final AttributeValue<?> value;
+        private final Class<? extends Factory> factory;
+
+        public Serializable(AttributeValue<?> value, Class<? extends Factory> factory) {
+            this.value = value;
+            this.factory = factory;
+        }
+
+        public Class<? extends Factory> getFactory() {
+            return factory;
+        }
+
+        public AttributeValue<?> getValue() {
+            return value;
+        }
+    }
+
+    public interface Factory {
+        Optional<QueueSerializable> deserialize(Serializable serializable);
```

Comment:
A `factory` deserialize a `serializable` to an optional of `serializable`. Seem not fluent. I mean we usually have a `Deserializer` deserialize a `serializable` to something but definitely not a `serializable`

Reply:
I've take another approach I've implemented at the beginning, it looks like C++, I don't it's more understandable, but at least it's more type correct.